pkg/controller/admissionchecks/provisioning/controller_test.go

```diff
@@ -516,7 +501,7 @@ func TestReconcile(t *testing.T) {
 			workload: baseWorkload.DeepCopy(),
 			checks:   []kueue.AdmissionCheck{*baseCheck.DeepCopy()},
 			flavors:  []kueue.ResourceFlavor{*baseFlavor1.DeepCopy(), *baseFlavor2.DeepCopy()},
-			configs:  []kueue.ProvisioningRequestConfig{*provReqConfigWithBaseBackoff(provReqConfigWithRetryLimit(baseConfig, 2), 0)},
+			configs:  []kueue.ProvisioningRequestConfig{*provReqConfigWithBaseBackoff(provReqConfigWithRetryLimit(baseConfig.DeepCopy(), 2), 0)},
```

Comment:
I imagine desired state would looks something like
```suggestion
			configs:  []kueue.ProvisioningRequestConfig{*baseConfig.DeepCopy().BaseBackoff(2).RetryLimit(0)},
```

Reply:
@PBundyra, So, the DeepCopy returns `ProvisioningRequestConfig` which doesnot have the `BaseBackoff` method on it. So, should i implement a `DeepCopy` for wrapper too ??
![image](https://github.com/user-attachments/assets/07ac3612-b3ff-472f-b335-6e1b7484d99e)

Reply:
Please provide Clone() method to duplicate wrapper.

Reply:
An example https://github.com/kubernetes-sigs/kueue/blob/main/pkg/util/testing/wrappers.go#L82-L84.